app/javascript/packs/task.js

```diff
@@ -0,0 +1,35 @@
+require("datatables.net-bs4");
+
+$(document).ready(function () {
+
+  $('#tasks_table').DataTable({
+    info: false,
+    paging: false,
+    searching: false
+  });
+  $("body").on('change',"#Project",function(e) {
+    e.preventDefault();
+    $.ajax({ type:'GET', url: '/tasks.js', data: { project_id: $(this).val(),
```

Comment:
all doing same thing, simple add a class to all 4 select tags and use single event. For example add a class `task-filters` and register single event